### PR TITLE
JS: More improvements to access paths

### PR DESCRIPTION
**Pull request by asger-semmle**
Commit-by-commit review encouraged.

Improvements to access paths:
- Rely on data flow nodes instead of expressions
- ~~Handle destructuring assignments, that is, for `let { y } = x`, let `y` be an alias for `x.y` instead of being the root of its own access path.~~
- ~~Handle local aliasing in some cases, that is, for `let y = x.y`, let `y` be an alias for `x.y`.~~

Also adds `SsaExplicitDefinition.getRhsNode()` which is something I've often been missing even if not all definitions have a RHS node. In the future, we could possibly add more cases here, such as for postfix update expressions.

[Evaluation](https://git.semmle.com/asger/dist-compare-reports/tree/domitian.ti.semmle.com_1557973546377) is uneventful but as usual this is a stepping stone towards making access paths usable for call graph construction.

### CODE REVIEW

**Comment by asger-semmle** [comment]
The test failures revealed some problems with how I tried to handle local aliasing, so I've rolled back the last commit for now. Hopefully we can land the preceding commits and I'll retry the aliasing and destructuring in a separate PR.

File: javascript/ql/src/semmle/javascript/SSA.qll [hunk 1]
```diff
@@ -521,6 +521,22 @@ class SsaExplicitDefinition extends SsaDefinition, TExplicitDef {
...
+   * Gets the data flow node representing the incoming value assigned at this definition,
+   * if any.
+   */
+  DataFlow::Node getRhsNode() {
+    exists(VarDef def | def = getDef() |
+      result = def.getSource().flow()
+      or
+      exists(VarRef ref |
+        ref = getSourceVariable().getAReference() and
+        def.getTarget().(BindingPattern).getABindingVarRef() = ref and
```
Comment (xiemaisi) [inline javascript/ql/src/semmle/javascript/SSA.qll]: `getABindingVarRef` does a deep traversal of the binding pattern, so `{ x: { y: z } }` has binding var ref `z`; presumably that's not what we want here?

Comment (xiemaisi) [reply javascript/ql/src/semmle/javascript/SSA.qll]: Ah, wait, I see that `patternPropRead` takes this into account; nvm.

**Comment by xiemaisi** [comment]
LGTM, but needs conflict resolution.

**Comment by asger-semmle** [comment]
Removing the last commit fixed the tests but [broke the evaluation](https://git.semmle.com/asger/dist-compare-reports/tree/elagabalus.ti.semmle.com_1558231917249). I'll tinker with it a bit more.

**Pull request closed without merging**